### PR TITLE
Feature/importance in code

### PR DESCRIPTION
**Pull request by knordstrom**

### CODE REVIEW

File: library/neo4j.py [hunk 1]
```diff
@@ -132,12 +132,22 @@ def process_org_chart(self, org: list[Employee]):
...
+            [print(r) for r in results]   
             return [Employee(**record['ceo']) for record in results]
         
+    def get_employee_with_full_org_chart(self, email: str) -> Employee:
+        result = self.get_org_chart_below(email)
+        if result is None:
+            return None
+        managers = self.get_org_chart_above(email)
+        if len(managers) > 1:
+            result.manager = managers[1]
```
Comment (mshash8) [inline library/neo4j.py:148]: I noticed that you add the employee in the result when this function get_org_chart_above is called. Is that for a reason?

File: tests/integration/library/test_email_weaviate.py [hunk 2]
```diff
@@ -93,6 +107,7 @@ def test_email_save(self, service):
                 "name": "Someone Important"
             },
             "provider": "Google",
+            "provider": "Google",
```
Comment (mshash8) [inline tests/integration/library/test_email_weaviate.py]: This line seems to be repeated.

File: library/importance.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        if distance is None or distance == 0:
+            return 0
+        
+        if distance == 1 or distance == -1:
+            importance = 100
+        elif distance > 0:
+            importance = min(100, 25*distance)
+        else: # distance < 0
+            importance = max(0, 100 - 25*abs(distance))
+        return importance
```
Comment (mshash8) [inline library/importance.py:79]: I was able to understand the distance == 1 or -1 case, but didn't quite understand how the importance score was calculated for skip managers and skip reportees.

Comment (knordstrom) [reply library/importance.py:79]: It's basically arbitrary logic, which is why I didn't put too much testing on it. 

Your manager two above is generally not communicating with you directly, mostly just departmental announcements, so I considered it less important. But then as you get more senior it seemed weird not to consider communications important (though I rarely found anything from the CEO of the company to be useful 6 layers down). 

The number of other people on an email and the identity of those users will be features we'll need to pay attention to in the real decision tree (which needs to be an xgboost or something and we'll need to get some labeled training data). Like if there are 1000 recipients or it's part of some DL as opposed to a direct communication from someone more senior (I had a few exchanges with the CEO directly and with the CPO that were certainly high importance in my inbox).

File: library/neo4j.py (hunk 1 above)
Comment (knordstrom) [reply library/neo4j.py:148]: The org chart above is a linked list, the org chart below is a tree. In both functions, the first entry is you.

So if you are the CEO, your list above has a single entry, yourself. 

There should be some clarity on this structure in the tests.

File: tests/integration/library/test_email_weaviate.py (hunk 2 above)
Comment (knordstrom) [reply tests/integration/library/test_email_weaviate.py]: Good eye

File: library/importance.py (hunk 3 above)
Comment (mshash8) [reply library/importance.py:79]: Thank you for the explanation. That makes a lot of sense. I think the number of people on an email really has a huge role to play in importance.

File: library/neo4j.py (hunk 1 above)
Comment (mshash8) [reply library/neo4j.py:148]: Okay got it! Thank you.

**Pull request merged**